[FEAT/#68] 로그인 연동 및 토큰 매니저

2ghrms opened the pull request:
## #️⃣PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가
- [ ] 기능 삭제
- [x] 버그 수정
- [x] 의존성, 환경 변수, 빌드 관련 코드 업데이트

## 🪵반영 브랜치
ex) feat/login -> dev

## 📝작업 내용
ex) 로그인 시, 구글 소셜 로그인 기능을 추가했습니다.
- 로그인 기능 구현
- 토큰 매니저 구현 [참고](https://clumsy-seeder-416.notion.site/26b1197c19ed805182e4cb69c5fba491?source=copy_link)
- 디바이스 토큰 연결/해제 구현
- 로그아웃 구현

## ✅테스트 결과
ex) 베이스 브랜치에 포함되기 위한 코드는 모두 정상적으로 동작해야 합니다. 결과물에 대한 스크린샷, GIF, 혹은 라이브

## 💬리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

아 충돌 ;

2ghrms committed:
[FEAT/#68] Add authentication API integration for student, partner, and admin login

- Add CommonLoginRequestDto and StudentLoginRequestDto for API requests
- Add CommonLoginResponseDto and StudentLoginResponseDto for API responses
- Add AuthService interface with common and student login endpoints
- Add AuthRepository interface and AuthRepositoryImpl implementation
- Add StudentLoginUseCase and CommonLoginUseCase for business logic
- Add LoginModel for authentication data
- Add LoginViewModel with login state management
- Integrate token management with login flow

2ghrms committed:
[FEAT/#68] 토큰 관리 시스템 구현

- TokenManager 클래스 추가
- SharedPreferences를 이용한 토큰 저장/조회/삭제 기능
- 자동 로그인 체크 기능
- Bearer 토큰 인터셉터 연동

2ghrms committed:
[FEAT/#68] 네트워크 설정 및 의존성 주입 모듈 구성

- ServiceModule에 Auth, NoAuth Qualifier 추가
- RepositoryModule 추가
- BearerInterceptor 토큰 자동 추가 기능
- 인증/비인증 API 클라이언트 분리

2ghrms committed:
[FEAT/#68] 로그인 UI 및 액티비티 수정

- LoginActivity에 실제 API 호출 로직 연동
- LmsLoginActivity WebView 로그인 후 서버 API 호출
- 로딩 UI 개선 (투명 배경, 파란색 스피너)
- 자동 로그인 체크 기능 추가
- 역할별 메인 액티비티 네비게이션 구현

2ghrms committed:
[FEAT/#68] 빌드 설정 및 네트워크 보안 구성

- BASE_URL을 실제 서버(https://assu.shop/)로 설정
- network_security_config.xml에 assu.shop 도메인 추가
- MyApplication 네트워크 상태 체크 로직 개선
- apiHandler 에러 처리 및 로깅 추가

2ghrms committed:
[FEAT/#68] 기타 서비스 API 연동 및 네트워크 설정

- ChattingService, DeviceTokenService BaseResponseDto 사용으로 통일
- DeviceTokenRepositoryImpl 타입 추론 에러 수정
- RetrofitResult 타입 통일 (util.RetrofitResult 사용)
- apiHandler 네트워크 에러 처리 개선

2ghrms committed:
[FEAT/#68] 공통 모델 및 유틸리티 추가

- RetrofitResult 공통 응답 모델 추가
- API 응답 처리를 위한 공통 타입 정의

eeeeeaaan commented on app/build.gradle.kts:
```diff
@@ -93,36 +89,4 @@ dependencies {
 
     // Kakao
     implementation("com.kakao.maps.open:android:2.12.8")
-
```
여기는 왜 지우신거에여??

2ghrms replied:
아니 이거 build.gradle이 갑자기 날라갔어요 ;; 잘못 지워져서 드갔습니다

2ghrms committed:
[REFACTOR/#54] Device Token API 엔드포인트 수정 및 tokenId 관리 추가

2ghrms committed:
[REFACTOR/#68] DeviceToken Repository 구조 정리 및 Hilt 바인딩 추가

2ghrms committed:
[REFACTOR/#68] TokenManager에 deviceTokenId 저장/조회 기능 추가

2ghrms committed:
[REFACTOR/#68] 로그아웃 시 서버 API 호출 및 토큰 삭제 플로우 개선

2ghrms committed:
[REFACTOR/#68] MainActivity에 @AndroidEntryPoint 추가로 Hilt 의존성 주입 활성화

2ghrms committed:
[REFACTOR/#68] MainActivity에 @AndroidEntryPoint 추가로 Hilt 의존성 주입 활성화

2ghrms committed:
[REFACTOR/#68] Device Token API 엔드포인트 수정 및 tokenId 관리 시스템 구축

leesumin0526 commented on app/src/main/java/com/example/assu_fe_app/data/dto/auth/BaseResponseDto.kt:
BaseResponse [data > dto]에 이미 있었는데 그걸 옮기신 건가요? 아님 추가하신건가요??
이미 다들 [data > dto]에 있던 거 쓰고 있었어서 새로 추가하신 거면 지우셔야 할 것 같아요

2ghrms replied:
제거했슴다

leesumin0526 commented on app/src/main/java/com/example/assu_fe_app/data/repository/deviceToken/DeviceTokenRepository.kt:
연결 부분(repo, usecase, model 등)은 develop에 제가 머지 해놨던 걸로 쓰셔야 할 것 같아요
사용되는 부분들 fragment, viewmodel은 오빠 거로 해도 될 것 같은데.. 확인은 해보셔야 할 것 같아요

leesumin0526 commented on app/src/main/java/com/example/assu_fe_app/fcm/MyFirebaseMessagingService.kt:
여기서 Activity 이동은 왜 하는 거에여??
필요하신 거면 제가 TTS 작업한 거랑 아마 충돌 날 것 같은데 잘 보시구 해결 부탁드립니다!!

2ghrms replied:
이거 알림 나중에 관리자랑 사용자에도 넣을까봐 호환성있게 넣어놨어요!!

2ghrms replied:
일단 이전 버전으로 돌려놓을게요

leesumin0526 commented on app/src/main/java/com/example/assu_fe_app/presentation/common/login/LoginViewModel.kt:
[presentation > common > mypage > MypageViewModel.kt]에 관련 내용 중복되는 것 같아여..ㅠㅠㅠㅠ

2ghrms replied:
엄 이것도 나중에 리팩토링해야할 것 같아요。。。 일단 동작하게 만들어둘게요 정말 죄송합니다 ㅠ

leesumin0526 commented on app/src/main/java/com/example/assu_fe_app/data/manager/TokenManager.kt:
제가 DeviceTokenStore를 따로 파서 디바이스 토큰이 저장되게 해놨었는데 확인해보시구 제 거 지우셔도 돼요
지우실 거면 그거 사용하고 있는 곳도 수정 부탁드려요...ㅠㅠㅠ...

2ghrms replied:
일단 충돌 해결하고 빨리 배포하는게 중요할 것 같아서 나중에 리팩토링으로 올려요.. 있는지 몰랐네요 죄송합니다

2ghrms replied:
일단 둘다 작동하게 해두겠습니다

leesumin0526 reviewed (commented):
고생하셨어요....

2ghrms committed:
Merge branch 'develop' of https://github.com/S0ftwareC0ntest/ASSU_FE_app into refactor/#54-modified-signup-login-ui

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/example/assu_fe_app/data/DevBearerInterceptor.kt
#	app/src/main/java/com/example/assu_fe_app/data/repository/deviceToken/DeviceTokenRepository.kt
#	app/src/main/java/com/example/assu_fe_app/data/repositoryImpl/deviceToken/DeviceTokenRepositoryImpl.kt
#	app/src/main/java/com/example/assu_fe_app/data/service/deviceToken/DeviceTokenService.kt
#	app/src/main/java/com/example/assu_fe_app/di/RepositoryModule.kt
#	app/src/main/java/com/example/assu_fe_app/domain/usecase/deviceToken/UnregisterDeviceTokenUseCase.kt
#	app/src/main/java/com/example/assu_fe_app/fcm/MyFirebaseMessagingService.kt
#	app/src/main/java/com/example/assu_fe_app/presentation/admin/mypage/AdminMypageFragment.kt
#	app/src/main/java/com/example/assu_fe_app/presentation/common/login/LoginActivity.kt
#	app/src/main/java/com/example/assu_fe_app/presentation/partner/mypage/PartnerMypageFragment.kt
#	app/src/main/java/com/example/assu_fe_app/presentation/user/mypage/UserMypageFragment.kt
#	app/src/main/java/com/example/assu_fe_app/ui/deviceToken/DeviceTokenViewModel.kt

2ghrms committed:
[FEAT/#68] 로그인 시스템 전면 개선 및 안정성 향상

- 로그인 성공 후 매끄러운 화면 전환을 위해 finish() 호출 추가
- FCM 토큰 등록 실패 시 앱 크래시 방지 (finish() 제거)
- 로그인 실패 시 사용자 친화적 에러 메시지 개선
- HTTP 상태 코드별 적절한 에러 처리 추가
- Dagger/Hilt 의존성 주입 문제 해결 (Retrofit Qualifier 추가)
- RepositoryModule을 RepoModule로 리네이밍
- 토큰 관리 및 인증 관련 로직 개선
- 디바이스 토큰 등록/해제 기능
- FCM 서비스 및 로그아웃 기능 개선

sjhwang0 reviewed (approved):
수고하셨습니다!

leesumin0526 reviewed (approved):
코드들 꼼꼼히 봤는데 잘 해결하셨네요!! 제가 바로 머지하겠습니다

leesumin0526 committed:
[REFACTOER/#54] Role 에러 수정
